pkg/validation/spec/gnostic.go

```diff
@@ -0,0 +1,1226 @@
+package spec
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/go-openapi/jsonreference"
+	openapi_v2 "github.com/googleapis/gnostic/openapiv2"
+)
+
+// Interfaces
+type GnosticCommonValidations interface {
+	GetMaximum() float64
+	GetExclusiveMaximum() bool
+	GetMinimum() float64
+	GetExclusiveMinimum() bool
+	GetMaxLength() int64
+	GetMinLength() int64
+	GetPattern() string
+	GetMaxItems() int64
+	GetMinItems() int64
+	GetUniqueItems() bool
+	GetMultipleOf() float64
+}
+
+type GnosticCommonValidationsNamedEnum interface {
+	GetEnum() []*openapi_v2.NamedAny
+}
+
+type GnosticCommonValidationsAnyEnum interface {
+	GetEnum() []*openapi_v2.Any
+}
+
+
+func (k *CommonValidations) FromGnostic(g GnosticCommonValidations) error {
+	if g == nil {
+		return nil
+	}
+
+	max := g.GetMaximum()
+	if max != 0 {
+		k.Maximum = &max
+	}
+
+	k.ExclusiveMaximum = g.GetExclusiveMaximum()
+
+	min := g.GetMinimum()
+	if min != 0 {
+		k.Minimum = &min
+	}
+
+	k.ExclusiveMinimum = g.GetExclusiveMinimum()
+
+	maxLen := g.GetMaxLength()
+	if maxLen != 0 {
+		k.MaxLength = &maxLen
+	}
+
+	minLen := g.GetMinLength()
+	if minLen != 0 {
+		k.MinLength = &minLen
+	}
+
+	k.Pattern = g.GetPattern()
+
+	maxItems := g.GetMaxItems()
+	if maxItems != 0 {
+		k.MaxItems = &maxItems
+	}
+
+	minItems := g.GetMinItems()
+	if minItems != 0 {
+		k.MinItems = &minItems
+	}
+
+	k.UniqueItems = g.GetUniqueItems()
+
+	multOf := g.GetMultipleOf()
+	if multOf != 0 {
+		k.MultipleOf = &multOf
+	}
+
+	if anyEnum, ok := g.(GnosticCommonValidationsAnyEnum); ok {
+		enums := anyEnum.GetEnum()
+
+		if len(enums) > 0 {
```

Comment:
is `enum: []` valid or distinct from `enum: null`?

Reply:
I've gone through to correct instance where len was checked in case of null, so that an instance of our object is always created if the gnostic version is also non-nil.